cvat-ui/src/containers/annotation-page/top-bar/top-bar.tsx

```diff
@@ -414,20 +457,46 @@ class AnnotationTopBarContainer extends React.PureComponent<Props, State> {
     };
 
     private onChangePlayerSliderValue = (value: number): void => {
-        const { playing, onSwitchPlay } = this.props;
+        const {
+            playing,
+            jobInstance,
+            frameNumber,
+            onSwitchPlay,
+            showDeletedFrames,
+            searchNonDeletedFrame,
+        } = this.props;
         if (playing) {
             onSwitchPlay(false);
         }
-        this.changeFrame(value as number);
+        if (showDeletedFrames) {
+            this.changeFrame(value);
+        } else {
+            const frameTo = value >= frameNumber ? jobInstance.stopFrame : jobInstance.startFrame;
+            const frameToBackward = value >= frameNumber ? jobInstance.startFrame : jobInstance.stopFrame;
+            searchNonDeletedFrame(value, frameTo, frameToBackward);
+        }
     };
 
     private onChangePlayerInputValue = (value: number): void => {
-        const { onSwitchPlay, playing, frameNumber } = this.props;
+        const {
+            jobInstance,
+            frameNumber,
+            onSwitchPlay,
+            playing,
+            showDeletedFrames,
+            searchNonDeletedFrame,
+        } = this.props;
         if (value !== frameNumber) {
             if (playing) {
                 onSwitchPlay(false);
             }
-            this.changeFrame(value);
+            if (!showDeletedFrames) {
+                const frameTo = value >= frameNumber ? jobInstance.startFrame : jobInstance.stopFrame;
+                const frameToBackward = value >= frameNumber ? jobInstance.stopFrame : jobInstance.startFrame;
+                searchNonDeletedFrame(value, frameTo, frameToBackward);
```

Comment:
According to our HLD we should not search not deleted frame if a user wants to get a specific frame using this field? Isn't it?

I also think that there is very similar case with slider above, also do not need to find not deleted frames. But it may be discussable.